gcp_deepvariant_runner.py

```diff
@@ -166,6 +168,62 @@
 }}
 """
 
+# Following const values are used to automatically set the computational flags.
+_WGS_MODEL_IDENTIFIER = 'wgs_standard'
+_WES_MODEL_IDENTIFIER = 'wes_standard'
+_WES_LARGE_THR = 12 * 1024 * 1024 * 1024
+_WGS_LARGE_THR = 200 * 1024 * 1024 * 1024
+_WGS_SMALL_THR = 25 * 1024 * 1024 * 1024
+
+
+class BamCategories(enum.Enum):
+  """List of BAM categories that determine automatically assigned flags."""
+  WES_SMALL = 0
+  WES_LARGE = 1
+  WGS_SMALL = 2
+  WGS_MEDIUM = 3
+  WGS_LARGE = 4
+
+
+# Default optimal computational flag values, one per BAM category.
+_DEFAULT_FLAGS = {}
+_DEFAULT_FLAGS[BamCategories.WES_SMALL] = {
+    'make_examples_workers': 8,
+    'make_examples_cores_per_worker': 1,
+    'call_variants_workers': 1,
+    'call_variants_cores_per_worker': 4,
+    'gpu': True
+}
+_DEFAULT_FLAGS[BamCategories.WES_LARGE] = {
+    'make_examples_workers': 16,
+    'make_examples_cores_per_worker': 1,
+    'call_variants_workers': 2,
+    'call_variants_cores_per_worker': 4,
+    'gpu': True
+}
+_DEFAULT_FLAGS[BamCategories.WGS_SMALL] = {
+    'make_examples_workers': 32,
+    'make_examples_cores_per_worker': 1,
+    'call_variants_workers': 4,
+    'call_variants_cores_per_worker': 4,
+    'gpu': True
+}
+_DEFAULT_FLAGS[BamCategories.WGS_MEDIUM] = {
+    'make_examples_workers': 64,
+    'make_examples_cores_per_worker': 1,
+    'tpu': True
+}
+_DEFAULT_FLAGS[BamCategories.WGS_LARGE] = {
+    'make_examples_workers': 128,
```

Comment:
I agree this is not scalable for batch runs of DV. We should emphasize in our documentations this flag should only be used for single runs. For batch runs, even 32 workers does not scale for 100s of input BAM files.
I believe what we offer through this flag will be very helpful for first time users of DV. When a customer reaches a point where they feel comfortable to run DV for 100s of BAM file, hopefully they have enough experience to set these flags optimally based on their specific needs. For example:
- If cost is the main deciding factor for a customer, still the most cost efficient solution is to run DV using 128 workers, sequentially one BAM file at a time. This might take weeks to finish 100s of samples.
- For another customer, weeks of processing might not be acceptable, so they will be running more expensive runs (using fewer workers with more cores) in parallel to finish their batch job within acceptable time frame.
I really don't think we should/can make that decision for our customers.